app/components/Selection.jsx

```diff
@@ -0,0 +1,103 @@
+var React = require('react')
+var Select = require('react-select')
+var History = require('react-router').History
+
+require('react-select/dist/default.css')
+
+var Selection = React.createClass({
+  mixins: [ History ],
+  getInitialState: function() {
+    return {hostTarget: '', clusterTarget: ''}
+  },
+  onHostSelectionChanged: function() {
+    this.setState({ hostTarget: this.refs.host.value })
+  },
+  onClusterSelectionChanged: function(clusterName) {
+    this.setState({ clusterTarget: clusterName })
+  },
+  getClusterValue: function() {
+    return this.state.clusterTarget
+  },
+  shouldDisableSubmit: function() {
+    return this.state.hostTarget == '' && this.state.clusterTarget == ''
+  },
+  onSubmit: function() {
+    var clusterComponents = this.state.clusterTarget.split('/')
+    var clusterName = clusterComponents[0]
+    var clusterProvider = clusterComponents[1]
+
+    // Route to our monitoring endpoint depending on whether we're in host
+    // or cluster mode.  This shit is pretty ugly, but it's what we have to
+    // do if we want to use a button instead of a normal anchor tag.
+    var streamMode = 'host'
+    var clusterData = {}
+    if(this.state.hostTarget == '') {
+      streamMode = 'cluster'
+      clusterData = { n: clusterName, p: clusterProvider }
+    }
+
+    var streamData = {sm: streamMode, h: this.state.hostTarget, c: clusterData}
+    var streamDataEncoded = window.btoa(JSON.stringify(streamData))
+
+    this.history.pushState(null, `/monitor/${streamMode}/${streamDataEncoded}`, null)
```

Comment:
This is an artifact of `react-router`'s design choice to move from `transitionTo` and is apparently related to their use of `rackt/history`... but man, it's fucking ugly and really takes away from the *intent* of what the user is trying to do.